src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -521,13 +521,24 @@ public void completeTask(String taskId, AccessControlResponse accessControlRespo
             task.setState(CFTTaskState.COMPLETED);
 
             //check the state, if not complete, complete
+            completeCamundaTask(taskId, taskHasCompleted);
+            //Commit transaction
+            cftTaskDatabaseService.saveTask(task);
+        }
+    }
+
+    private void completeCamundaTask(String taskId, boolean taskHasCompleted) {
+        try {
+            //Perform Camunda updates
+            camundaService.completeTask(taskId, taskHasCompleted);
+        } catch (Exception e) {
```

Comment:
You probably only want to do this when there's an error that indicates it could have already been completed.  E.g. 404 means the Task does not exist in the active table so we can check history.  
Other errors may not need this check e.g. 500 means there's a Camunda error so just abandon, 400 error means something is wrong on our side so we probably can't update Camunda.